lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2154,7 +2154,7 @@ bool llvm::writeSpirv(Module *M, const SPIRV::TranslatorOpts &Opts,
 
   legacy::PassManager PassMgr;
   addPassesForSPIRV(PassMgr);
-  if (hasLoopUnrollMetadata(M))
+  if (hasLoopMetadata(M))
```

Comment:
[Optional] It might be good to have a comment why we need loopsimplify pass here.

Reply:
Indeed. @vmaksimo please add a comment explaining why we need to simplify loops.